src/pages/node-express/lesson2-a1.mdx

```diff
@@ -0,0 +1,304 @@
+  order to get the actual return value. To resolve a Promise inside an{" "}
+  <code>async function</code>, you use the keyword, <code>await</code>. This
+  should be used inside a <code>try/catch</code> block so that you can handle
+  any errors, as follows:
+</p>
+
+```js
+const myFunc = async () => {
+    ...
+    return result
+}
+
+const myFunc2 = async () => {
+    try {
+       result = await myFunc()
+       ...
+    } catch(err) {
+        console.log("An error occurred: ", err)
+    }
+}
+```
+
+<p class="ml-8">
+  Sometimes you need to call an async function from within another function that
+  is not async. In this case, you can’t use <code>await</code> – it will give a
+  syntax error. So you can either use <code>.then</code> or you can wrap the
+  function call as follows:
+</p>
+
+```js
+const myFunc3 = () => {  // not async, and in some contexts we better not make it async
+   myFunc()
+   .then((result) => {
+      console.log("got the result.")
+      ...
+   })
+   .catch((error)=> {
+      console.log("An error occurred: ", error)
+   })
+}
+
+const myFunc4 = () => {  // the other way to do it, via a wrapper:
+    const myFunc5 = async () => {
+        try {
+               result = await myFunc() {
+               console.log("got the result.")
+               ...
+            }
+        } catch(error) => {
+            console.log("An error occurred: ", error)
+        }
+    }
+    myFunc5() // here's where we call the wrapper, but do NOT do this:
+    // result = myFunc5()  This won't work because myFunc5 is asynchronous!
+    // All you'd get back is a Promise, not the result.
+}
+```
+
+<p class="ml-8">
+  There’s one more trick with the <code>.then</code>. Suppose you need to make a
+  string of calls to async functions. You can chain the <code>.then</code>{" "}
+  statements as follows:
+</p>
+
+```js
+const myFunc6 = () => {
+  myFunc() // an async function, so it returns a promise
+    .then((result) => {
+      console.log("got the first result");
+      return myFunc(); // here we call it again, we return the promise myFunc returns
+    })
+    .then((result) => {
+      console.log("got the second result");
+    })
+    .catch((err) => {
+      console.log("An error occurred: ", err);
+    });
+};
+```
+
+<p class="ml-8">
+  So, you can chain a collection of async calls with <code>.then</code>{" "}
+  statements, followed by one <code>.catch</code> at the end.
+</p>
+
+Ok, that’s the summary. Be sure that you understand this. We will do a lot of asynchronous programming in Express.
+
+---
+
+Now for the programs to write for this assignment:
+
+1.  Create a program named `writeWithPromisesAwait.js`. We are going to use the fs.promises package. `fs` is the built-in "File system" set of functions in [Node](https://nodejs.org/api/fs.html#promises-api). By adding `.promises` we're going to access the versions of those built-in functions that return a Promise as their result. You’d start with the following code:
```

Comment:
Added a bit more of a blurb/intro for using this `.promises` export of the `fs` module